src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -723,9 +755,33 @@ export class EditorComponent implements OnInit {
   }
 
   /**
-   *  catches emitted MaskAction from mask.directive and calls the undo/redo 'do' function.
+   *  Catches emitted event from mask.directive once users mouse lifts up.
+   *  Finds all pixels painted on paint canvas and adds to set to pass into maskController.
+   *  Calls the undo/redo 'do' function with paintedMask: Set of imageData indexes.
+   *  TODO: Pass in four pixels that represent the <X, >X, <Y, >Y to not traverse over entire data array
    */
-  newMaskController(maskAction: MaskAction) {
+
+  maskControllerPaint() {
+    let paintedImageData = this.paintCtx.getImageData(0, 0,this.image.width, this.image.height).data;
+    let paintedMask = new Set<number>();
+
+    let leftTop = this.searchRectangle.getLeftTop();
+    let bottomRight = this.searchRectangle.getBottomRight();
+
+    let leftTopIndex = this.magicWandService.coordToDataArrayIndex(leftTop.x, leftTop.y, this.image.width);
+    let bottomRightIndex = this.magicWandService.coordToDataArrayIndex(bottomRight.x, bottomRight.y, this.image.width);
```

Comment:
These variables don't seem to be used?

Also, they should be `const`.

Reply:
Sorry, I should have added TODOs in here, I wanted to have you both review it sooner and then I was going to push a second review with it actually implemented. I will work on implementing it now.

Reply:
For the sake of getting this PR in (and having smaller PRs in general), then let's go ahead and put the TODO comment in for now, and have this part implemented in a follow-up PR.